### ISSUE TITLE
move_directory does not try to rename directory when destination does not exist

### ISSUE THREAD

**Issue by tkkcc** [simongoricar/fs-more#1]
like make this
https://github.com/simongoricar/fs-more/blob/3e0516202e71c036e12e7f405c61f244960ce929/src/directory/move.rs#L167-L169

to this
```rust
    if validated_destination_directory.state == DestinationDirectoryState::IsNotEmpty {
        return Ok(DirectoryMoveByRenameAction::Impossible);
    }
```

**Comment by simongoricar** [comment]
Interesting, thanks. Seems like an oversight, leading to the windows branch below it to never succeed? 

Will test and update so it tries to rename when the destination doesn't exist or is empty (the latter already works). Might be a little bit before I get to it though.

**Comment by simongoricar** [comment]
I've moved the conditions for renaming to destination into each of the platform-specific branches (see e8e5a6ef1498c5ea7ce8925d866a8a954ed38542). It should hopefully be slightly clearer what the prerequisites are, and this bug should be fixed. 

I've also added a few tests that cover this in b988e643f15e041447514949959f469cfccdf4c7.

**Comment by simongoricar** [comment]
The fix for this has landed in `v0.7.0`, which will be available on crates.io shortly. Thanks for the report!
